### PR TITLE
sdk/js-query

### PR DESCRIPTION
**Pull request by kev1n-peters**

### CODE REVIEW

File: sdk/js-query/.gitignore [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# testing
+/coverage
+
+# production
+/build
+
+# misc
+.DS_Store
+.env.local
```
Comment (panoel) [inline sdk/js-query/.gitignore:16]: Do you want to do `.env.*`?

File: sdk/js-query/LICENSE [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Copyright 2020 Wormhole Project Contributors
```
Comment (panoel) [inline sdk/js-query/LICENSE:1]: You've been working on this for awhile.

File: sdk/js-query/src/query/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export * from "./request";
+export * from "./response";
+export * from "./utils";
+export * from "./ethCall";
```
Comment (panoel) [inline sdk/js-query/src/query/index.ts:4]: Did you need to add ./consts?

File: sdk/js-query/src/query/utils.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export function isHex(s: string) {
+  // TODO: even length?
+  return /^(0x)?[0-9a-fA-F]+$/.test(s);
+}
+
+export function hexToUint8Array(s: string): Uint8Array {
+  if (!isHex(s)) {
+    throw new Error(`${s} is not hex`);
+  }
+  return new Uint8Array(
```
Comment (panoel) [inline sdk/js-query/src/query/utils.ts:12]: You may need to prepend a '0' if the length is odd?  That threw me off in the algorand sdk.

File: sdk/js-query/src/query/utils.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    Buffer.from(s.startsWith("0x") ? s.slice(2) : s, "hex")
```
Comment (panoel) [inline sdk/js-query/src/query/utils.ts:13]: May need to append a leading '0' if the length is odd:
```> let b = Buffer.from("fff", "hex")
undefined
> b
<Buffer ff>
> let bb = Buffer.from("0fff", "hex")
undefined
> bb
<Buffer 0f ff>```

File: sdk/js-query/src/query/response.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { ChainQueryType } from "./request";
+
+export class QueryResponse {
```
Comment (panoel) [inline sdk/js-query/src/query/response.ts:3]: This whole file is TODO?

**Review by panoel — DISMISSED** [review]

**Review by bruce-riley — APPROVED** [review]

File: sdk/js-query/src/query/utils.ts (hunk 5 above)
Comment (kev1n-peters) [reply sdk/js-query/src/query/utils.ts:15]: We assert that the length is even now in `isValidHexString`

File: sdk/js-query/src/query/response.ts (hunk 6 above)
Comment (kev1n-peters) [reply sdk/js-query/src/query/response.ts:5]: Yeah, still very much WIP

**Review by bruce-riley — APPROVED** [review]

**Review by panoel — APPROVED** [review]

**Pull request merged**